wikimedia/wikipedia-ios
# Change Components and WKData to local packages

**Phabricator:** https://phabricator.wikimedia.org/T345645 (First item)

### Notes
This PR re-establishes our monorepo setup, to remove juggling between 3 repositories. I think I have managed to remove any concerns with local package revision IDs as well, because there are no more references to them in the package manifest files.

Most of this setup came from following instructions in https://developer.apple.com/documentation/xcode/organizing-your-code-with-local-packages. Then I copied over the Source and Test files into the local packages.

Note: So far PR tests only run against the main app unit tests, but I think I can modify the workflow to include Components and WKData tests in there as well. I don't want to change that workflow until this is merged.

### Test Steps
1. Run app.
2. Make a breaking change in WKData and Components. Confirm compile errors bubble up to app call sites.
3. Confirm you can fix errors in app call sites, and see changes reflected across all 3 areas in git.
